plugins/GSdx/Renderers/OpenGL/GLLoader.cpp

```diff
@@ -332,28 +333,32 @@ namespace GLLoader {
 			fprintf(stderr, "Error: GLLoader failed to get GL version\n");
 			return false;
 		}
-		GLuint v = 1;
-		while (s[v] != '\0' && s[v-1] != ' ') v++;
-
 		const char* vendor = (const char*)glGetString(GL_VENDOR);
 		if (s_first_load)
-			fprintf(stdout, "OpenGL information. GPU: %s. Vendor: %s. Driver: %s\n", glGetString(GL_RENDERER), vendor, &s[v]);
+			fprintf(stdout, "OpenGL information. GPU: %s. Vendor: %s. Driver: %s\n", glGetString(GL_RENDERER), vendor, &s[0]);
 
-		// Name changed but driver is still bad!
-		if (strstr(vendor, "Advanced Micro Devices") || strstr(vendor, "ATI Technologies Inc.") || strstr(vendor, "ATI"))
+		// 32-bit driver should always actually just report the former
+		if (strstr(vendor, "ATI Technologies Inc.") || strstr(vendor, "Advanced Micro Devices") || strstr(vendor, "ATI")) {
 			vendor_id_amd = true;
-		/*if (vendor_id_amd && (
-				strstr((const char*)&s[v], " 10.") || // Blacklist all 2010 AMD drivers.
-				strstr((const char*)&s[v], " 11.") || // Blacklist all 2011 AMD drivers.
-				strstr((const char*)&s[v], " 12.") || // Blacklist all 2012 AMD drivers.
-				strstr((const char*)&s[v], " 13.") || // Blacklist all 2013 AMD drivers.
-				strstr((const char*)&s[v], " 14.") || // Blacklist all 2014 AMD drivers.
-				strstr((const char*)&s[v], " 15.") || // Blacklist all 2015 AMD drivers.
-				strstr((const char*)&s[v], " 16.") || // Blacklist all 2016 AMD drivers.
-				strstr((const char*)&s[v], " 17.") // Blacklist all 2017 AMD drivers for now.
-				))
-			amd_legacy_buggy_driver = true;
-		*/
+			// AMD's GL_version is made of: <OpenGL version number> <requested context> <DriverVersion>
+			// Technically speaking we'd just be interested in the atioglxx/fglrx_dri userspace lib
+			// (whose revision number is comfortably exposed in the first element of the string above)
+			// Unfortuntately after testing for at least these 2 cases we check, I found that had same
+			// versions across the buggy and fixed "branch". Threfore look for the kernel driver build.
+			//
+			// Screw Wx screwings (it's nice we have decimal seprator localized... but dlls do not)
```

Comment:
Pointless WX comment.

There are some typos.

Reply:
I see why it's pointless, but at the same time I'm really not fond of that code. 
Be it the different windows (msvc?) and linux functions, or failing to overcome Wx forcing otherwise.